modules/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java

```diff
@@ -196,12 +198,21 @@ private static class AzureHTTPStatsCollectorHandler extends HttpStatsCollectorHa
         private static final Predicate<String> LIST_PATTERN = Pattern.compile("GET /[a-zA-Z0-9]+/[a-zA-Z0-9]+\\?.+").asMatchPredicate();
         private static final Predicate<String> GET_BLOB_PATTERN = Pattern.compile("GET /[a-zA-Z0-9]+/[a-zA-Z0-9]+/.+").asMatchPredicate();
 
+        private final Set<String> seenRequestIds = ConcurrentCollections.newConcurrentSet();
+
         private AzureHTTPStatsCollectorHandler(HttpHandler delegate) {
             super(delegate);
         }
 
         @Override
         protected void maybeTrack(String request, Headers headers) {
+            // Same request id is a retry
+            // https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-ncnbi/817da997-30d2-4cd3-972f-a0073e4e98f7
+            // Do not count retries since the client side request stats do not track them yet.
```

Comment:
For the original API that tracked requests we shouldn't track retries since that API is used for billing and failed requests are not charged.

Reply:
Hmm we are counting retries for s3 for the [API](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-repositories-metering-api.html). So one of them is incorrect.

Reply:
I think you're right. We're counting the retries for S3 which is slightly incorrect for the billing purposes